Confusing error messages, cannot generateObject with grok-4-fast-reasoning

### Description

```
const model = createXai({
					apiKey: this.env.XAI_API_KEY,
				});
				const { object } = await generateObject({
					model: model.languageModel('grok-4-fast-reasoning'),
					schema: z.object({
						name: z.string(),
					}),
					system: 'Identify the name of the person from the following text',
					messages: [{ role: 'user', content: 'My name is John Doe' }],
					providerOptions: {
						xai: {
							reasoningEffort: 'high',
						} satisfies XaiProviderOptions,
					},
				});
```

Throws a `[ERROR] Bad Request` error. However if i comment out the `providerOptions`, i get another error: `[ERROR] AI SDK Warning: The "responseFormat" setting is not supported by this model - JSON response format schema is not supported`

### AI SDK Version

"@ai-sdk/xai": "^2.0.23",
		"ai": "^5.0.61",

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/.changeset/many-tips-clap.md
+++ b/.changeset/many-tips-clap.md
@@ -1,0 +1,5 @@
+---
+'@ai-sdk/xai': patch
+---
+
+fix(provider/xai): remove json schema unsupported warning

--- a/examples/ai-core/src/generate-object/xai-grok-3-mini-reasoning-effort.ts
+++ b/examples/ai-core/src/generate-object/xai-grok-3-mini-reasoning-effort.ts
@@ -1,0 +1,35 @@
+import { xai } from '@ai-sdk/xai';
+import { generateObject } from 'ai';
+import 'dotenv/config';
+import { z } from 'zod';
+
+async function main() {
+  const result = await generateObject({
+    model: xai('grok-3-mini'),
+    schema: z.object({
+      name: z.string(),
+      age: z.number().optional(),
+      occupation: z.string().optional(),
+    }),
+    system: 'identify the person information from the following text',
+    messages: [
+      {
+        role: 'user',
+        content:
+          'my name is john doe, i am 35 years old and work as a software engineer',
+      },
+    ],
+    providerOptions: {
+      xai: {
+        reasoningEffort: 'high',
+      },
+    },
+  });
+
+  console.log('extracted person:', result.object);
+  console.log();
+  console.log('token usage:', result.usage);
+  console.log('finish reason:', result.finishReason);
+}
+
+main().catch(console.error);

--- a/examples/ai-core/src/generate-object/xai-grok-4-fast-reasoning.ts
+++ b/examples/ai-core/src/generate-object/xai-grok-4-fast-reasoning.ts
@@ -1,0 +1,30 @@
+import { xai } from '@ai-sdk/xai';
+import { generateObject } from 'ai';
+import 'dotenv/config';
+import { z } from 'zod';
+
+async function main() {
+  const result = await generateObject({
+    model: xai('grok-4-fast-reasoning'),
+    schema: z.object({
+      name: z.string(),
+      age: z.number().optional(),
+      occupation: z.string().optional(),
+    }),
+    system: 'identify the person information from the following text',
+    messages: [
+      {
+        role: 'user',
+        content:
+          'my name is john doe, i am 35 years old and work as a software engineer',
+      },
+    ],
+  });
+
+  console.log('extracted person:', result.object);
+  console.log();
+  console.log('token usage:', result.usage);
+  console.log('finish reason:', result.finishReason);
+}
+
+main().catch(console.error);

--- a/packages/xai/src/xai-chat-language-model.test.ts
+++ b/packages/xai/src/xai-chat-language-model.test.ts
@@ -719,6 +719,64 @@ describe('XaiChatLanguageModel', () => {
         ]
       `);
     });
+
+    it('should support json schema response format without warnings', async () => {
+      prepareJsonResponse({ content: '{"name":"john doe"}' });
+
+      const { warnings } = await model.doGenerate({
+        prompt: TEST_PROMPT,
+        responseFormat: {
+          type: 'json',
+          schema: {
+            type: 'object',
+            properties: {
+              name: { type: 'string' },
+            },
+            required: ['name'],
+            additionalProperties: false,
+          },
+        },
+      });
+
+      expect(warnings).toEqual([]);
+    });
+
+    it('should send json schema in response format', async () => {
+      prepareJsonResponse({ content: '{"name":"john"}' });
+
+      await model.doGenerate({
+        prompt: TEST_PROMPT,
+        responseFormat: {
+          type: 'json',
+          name: 'person',
+          schema: {
+            type: 'object',
+            properties: {
+              name: { type: 'string' },
+            },
+            required: ['name'],
+          },
+        },
+      });
+
+      expect(await server.calls[0].requestBodyJson).toMatchObject({
+        model: 'grok-beta',
+        response_format: {
+          type: 'json_schema',
+          json_schema: {
+            name: 'person',
+            schema: {
+              type: 'object',
+              properties: {
+                name: { type: 'string' },
+              },
+              required: ['name'],
+            },
+            strict: true,
+          },
+        },
+      });
+    });
   });
 
   describe('doStream', () => {

--- a/packages/xai/src/xai-chat-language-model.ts
+++ b/packages/xai/src/xai-chat-language-model.ts
@@ -105,18 +105,6 @@ export class XaiChatLanguageModel implements LanguageModelV3 {
       });
     }
 
-    if (
-      responseFormat != null &&
-      responseFormat.type === 'json' &&
-      responseFormat.schema != null
-    ) {
-      warnings.push({
-        type: 'unsupported-setting',
-        setting: 'responseFormat',
-        details: 'JSON response format schema is not supported',
-      });
-    }
-
     // convert ai sdk messages to xai format
     const { messages, warnings: messageWarnings } =
       convertToXaiChatMessages(prompt);

